lesson18/script.js

```diff
@@ -0,0 +1,186 @@
+
+const init = async () => {
+    showLoading();
+    try{
+        const sliderData = await fetchSliderData(3000);
+        if (!sliderData) {
+            return;
+        }
+        renderSliderElements(sliderData);
+        autoSwitchSlider(sliderData);
+    }catch(error){
+        console.error(error);
+        createErrorMessage(error);
+    }finally{
+        removeLoading();
+    }
+}
+
+const renderSliderElements = (sliderData) => {
+    renderSliderItems(sliderData);
+    getPageNumber(sliderData);
+    renderButtons(sliderData);
+    renderDotPagination(sliderData);
+    isActiveFirstItem();
+}
+
+let autoPlay;
+const autoSwitchSlider = (sliderData) =>{
+    autoPlay = setInterval(() =>{
+        currentIndex ++;
+        if(currentIndex >= sliderData.length){
+            currentIndex = 0;
+        }
+        switchSlider(sliderData,currentIndex);
+    }, 3000);
+}
+
+const resetAutoPlay = (sliderData) => {
+    clearInterval(autoPlay);
+    autoSwitchSlider(sliderData);
+}
+
+const renderSliderItems = (sliderData) => {
+    const fragment = document.createDocumentFragment();
+    for(const image of sliderData){
+        const sliderItem = createAttributedElements({tag:"div",attrObj:{class:"slider__item"}});
+        const slideImage = createAttributedElements({tag:"img",attrObj:{class:"slide-image",src:image.img,alt:image.alt}});
+        
+        fragment.appendChild(sliderItem).appendChild(slideImage);
+    }
+
+    slider.appendChild(fragment);
+}
+
+const getPageNumber = (sliderData) => {
+    const pageNumber    = createAttributedElements({tag:"p",attrObj:{class:"page-number"},str:`/${sliderData.length}`});
+    const currentNumber = createAttributedElements({tag:"span",attrObj:{id:"js-currentNumber",class:"current-number"},str:"1"});
+
+    slider.appendChild(pageNumber).insertAdjacentElement("afterbegin",currentNumber);
+}
+
+const renderButtons = (sliderData) => {
+    const prevBtn = createAttributedElements({tag:"button",attrObj:{id:"js-prevBtn",class:"js-btn prev-btn slider-btn",disabled:true}});
+    const nextBtn = createAttributedElements({tag:"button",attrObj:{id:"js-nextBtn",class:"js-btn next-btn slider-btn"}});
+
+    slider.appendChild(prevBtn).insertAdjacentElement("afterend",nextBtn);
+    addClickEventForButton(sliderData);
+}
+
+const renderDotPagination = (sliderData) => {
+    const paginationList = createAttributedElements({tag:"ul",attrObj:{id:"js-dot-pagination",class:"dot-pagination"}});
+
+    const fragment = document.createDocumentFragment();
+    const length   = sliderData.length;
+    for(let i = 0; i < length; i++){
+        const paginationItem = createAttributedElements({tag:"li",attrObj:{class:"dot-pagination__item"}});
+        fragment.appendChild(paginationItem);
+    }
+
+    slider.appendChild(paginationList).appendChild(fragment);
+    addClickEventForDotPagination(sliderData);
+}
+
+let currentIndex = 0;
```

Comment:
IMO
How about assigning to an object using const instead of using let outside the function? 
I got reviewed this way on the same assignment and I'm sharing!
[reference](https://github.com/csakyo/homework/pull/32#discussion_r857066877)

Reply:
Thank you for your comment!
I find objects useful when managing multiple properties.
However, it is unlikely that any other property will be added to currentIndex here.
So I'm confused as to whether there is a merit in managing only one thing with an object.

I think it is useful in the following cases.
const user = {
        Name: Marriage,
        Age: 20 years old
        Height: 154,
        Weight: 45
}

Reply:
I learned that it is better to manage with an object to avoid declaration with let as much as possible.
Thank you for learning a lot.
There was another thing declared by let, so I created an object called sliderStatus and put it together!
I fixed.
https://github.com/nakagawamai/js-lesson/pull/20/commits/aa72fcd0d811a7b777e813327770a99f2e5cba25